adapter/src/main/java/com/twilio/kudu/sql/rules/KuduSortRule.java

```diff
@@ -83,21 +84,24 @@ public boolean canApply(final RelTraitSet sortTraits, final KuduQuery query, fin
               && sortField.direction != RelFieldCollation.Direction.STRICTLY_ASCENDING)) {
         return false;
       }
-      // the sort columns must be a prefix of the primary key columns
-      if (sortField.getFieldIndex() >= openedTable.getSchema().getPrimaryKeyColumnCount()
-          || sortField.getFieldIndex() != pkColumnIndex) {
-        // This field is not in the primary key columns. If there is a condition lets
-        // see if it is there
-        if (filter.isPresent()) {
-          final RexNode originalCondition = filter.get().getCondition();
-          while (pkColumnIndex < sortField.getFieldIndex()) {
-            final KuduFilterVisitor visitor = new KuduFilterVisitor(pkColumnIndex);
-            final Boolean foundFieldInCondition = originalCondition.accept(visitor);
-            if (foundFieldInCondition.equals(Boolean.FALSE)) {
-              return false;
-            }
-            pkColumnIndex++;
+      if (sortField.getFieldIndex() >= openedTable.getSchema().getPrimaryKeyColumnCount()) {
```

Comment:
Previously if we were sorting on a non pk column with a filter applied ``(NON_PK_COL='X')`` this rule would be triggered, but with this change we won't apply this rule, is that right?
I think we should keep the previous comments since it helps in understanding the code.

Reply:
Interesting, I had forgotten about this feature. We can apply the sort for the following query

```sql
SELECT * FROM "SOME_TABLE" WHERE NON_PK1 = 'XYZ' ORDER BY PK1 ASC, NON_PK1 ASC;
```

I believe this PR removes that feature

Reply:
Added this back in this commit: https://github.com/twilio/calcite-kudu/pull/2/commits/93000fbf4aebfb94cfdd2264fdaf3a8df96246ca